README.md

```diff
@@ -269,58 +271,57 @@ onDialNumber(data) {
+Hubspot ensures the call widget is logged in before sending in a dial number event - if the widget is not logged in, an alert is shown in HubSpot's UI.
 
-The following messages are exchanged when a call widget is instantiated -
+Here is description of events:
 
-1. [HubSpot] Sends a SYNC message
-2. [SoftPhone] Sends SYNC_ACK message
-3. [SoftPhone] Sends INITIALIZED message with login state and optionally widget size
+1. **Dial number** - HubSpot sends the dial number event.
+2. **Outbound call started** - Widget notifies HubSpot when the call is started.
+3. **Create engagement** - HubSpot creates an engagement with minimum information if requested by the widget.
+4. **Engagement created** - HubSpot creates an engagement.
+5. **Engagement created** - HubSpot sends the engagementId to the widget.
+6. **Call ended** - Widget notifies when call is ended.
+7. **Call completed** - Widget notifies when user is done with the widget user experience.
+8. **Update engagement** - Widget fetches the engagment by the engagementId, merges and updates the engagement with additional call details. [Call engagement overview](https://developers.hubspot.com/docs/methods/engagements/engagements-overview), [Docs on updating the engagement](https://developers.hubspot.com/docs/methods/engagements/update_engagement-patch)
 
-At this point, the messages can be exchanged between the call widget and HubSpot.
+# FAQs
 
-Note that the SYNC message sent repeatedly until it receives a response from the iFrame to account for slow loading call widgets.
+### App
 
-### Outbound call
+<details>
+ <summary>How is user authentication handled?</summary>
+ <p>
+    The call widget should handle authentication.
+</p>
+</details>
 
-The following messages are exchanged when user initiates a call -
+<details>
+ <summary>Is Calling Extensions hosted on a CDN?</summary>
+ <p>
+    No. The calling entensions is very small and should be bundled with the call widget.  If bundling the file is not possible, the npm package includes a compiled UMD bundle that can be included into HTML (../node_modules/@hubspot/calling-extensions-sdk/dist/main.js).
+</p>
+</details>
 
-1. [HubSpot] Sends DIAL_NUMBER message with phone_number to dial
-2. [SoftPhone] Sends the OUTGOING_CALL_STARTED message with the phone number that is dialed
-3. [SoftPhone] Sends the CALL_ANSWERED message
-4. [SoftPhone] Sends the CALL_ENDED message
+<details>
+ <summary>When an engagement should be created vs updated.</summary>
+ <p>
+    A user can initiate a call from inside the HubSpot UI and outside of the HubSpot UI (e.g. mobile app/redirected number/etc.)  If a call is initiated from within HubSpot UI, HubSpot will create a call engagement and send the engagement to the call widget.  Once the call finishes, the call widget can update this engagement with additional call details.  If a call is initiated outside of HubSpot UI, the widget should create the call engagement. 
+</p>
```

Comment:
I'm curious about this. Above we mention that `callInfo` has a `createEngagement` property. How does that relate to "When an engagement should be created vs updated."?

Reply:
w/o the HubSpot UI those properties play no role - in that case the widget should create an engagement.  createEngagement is flag for our FE to create an engagement.